auth/src/main/java/com/firebase/ui/auth/ui/idp/AuthMethodPickerActivity.java

```diff
@@ -166,9 +166,11 @@ protected void onFailure(@NonNull Exception e) {
                 if (e instanceof FirebaseAuthAnonymousUpgradeException) {
                     finish(ErrorCodes.ANONYMOUS_UPGRADE_MERGE_CONFLICT,
                             ((FirebaseAuthAnonymousUpgradeException) e).getResponse().toIntent());
-                } else if ((!(e instanceof UserCancellationException))) {
-                    String text = e instanceof FirebaseUiException ? e.getMessage() :
-                            getString(R.string.fui_error_unknown);
+                } else if (e instanceof FirebaseUiException) {
+                    FirebaseUiException fue = (FirebaseUiException) e;
+                    finish(RESULT_CANCELED, IdpResponse.from(fue).toIntent());
+                } else {
```

Comment:
I think this might show an error message when the user presses back without `!(e instanceof UserCancellationException)`.

Reply:
Oops thank you!